compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -878,15 +885,22 @@ impl<'a> Linker for MsvcLinker<'a> {
     }
 
     fn link_dylib_by_name(&mut self, name: &str, verbatim: bool, _as_needed: bool) {
-        self.link_arg(format!("{}{}", name, if verbatim { "" } else { ".lib" }));
+        // On MSVC-like targets rustc supports static libraries using alternative naming
+        // scheme (`libfoo.a`) unsupported by linker, search for such libraries manually.
+        let path = find_native_dynamic_library(name, verbatim, self.sess);
+        self.link_arg(path);
     }
 
     fn link_dylib_by_path(&mut self, path: &Path, _as_needed: bool) {
         // When producing a dll, MSVC linker may not emit an implib file if the dll doesn't export
         // any symbols, so we skip linking if the implib file is not present.
-        let implib_path = path.with_extension("dll.lib");
-        if implib_path.exists() {
-            self.link_or_cc_arg(implib_path);
+        let formats = [".dll.lib", ".dll.a"];
+        for extension in formats {
```

Comment:
```suggestion
        for extension in ["dll.lib", "dll.a"] {
```
The dots are [unnecessary](https://doc.rust-lang.org/std/path/struct.Path.html#method.with_extension) (and maybe incorrect), also unnecessary local variable.

Reply:
Marking as resolved in light of https://github.com/rust-lang/rust/pull/123436#discussion_r1744137224